server/lib/search.js

```diff
@@ -151,42 +151,42 @@ export const searchCollectivesInDB = async (
   }
 
   if (hostCollectiveIds && hostCollectiveIds.length > 0) {
-    dynamicConditions += 'AND "HostCollectiveId" IN (:hostCollectiveIds) ';
+    dynamicConditions += 'AND c."HostCollectiveId" IN (:hostCollectiveIds) ';
   }
 
   if (isHost) {
-    dynamicConditions += `AND "isHostAccount" IS TRUE AND "type" = 'ORGANIZATION' `;
+    dynamicConditions += `AND c."isHostAccount" IS TRUE AND c."type" = 'ORGANIZATION' `;
   }
 
   if (types?.length) {
-    dynamicConditions += `AND "type" IN (:types) `;
+    dynamicConditions += `AND c."type" IN (:types) `;
   }
 
   if (onlyActive) {
-    dynamicConditions += 'AND "isActive" = TRUE ';
+    dynamicConditions += 'AND c."isActive" = TRUE ';
   }
 
   if (skipRecentAccounts) {
-    dynamicConditions += `AND (COALESCE(("data"#>>'{spamReport,score}')::float, 0) <= 0.2 OR "createdAt" < (NOW() - interval '2 day')) `;
+    dynamicConditions += `AND (COALESCE((c."data"#>>'{spamReport,score}')::float, 0) <= 0.2 OR c."createdAt" < (NOW() - interval '2 day')) `;
   }
 
   if (typeof hasCustomContributionsEnabled === 'boolean') {
     if (hasCustomContributionsEnabled) {
-      dynamicConditions += `AND ("settings"->>'disableCustomContributions')::boolean IS NOT TRUE `;
+      dynamicConditions += `AND (c."settings"->>'disableCustomContributions')::boolean IS NOT TRUE `;
     } else {
-      dynamicConditions += `AND ("settings"->>'disableCustomContributions')::boolean IS TRUE `;
+      dynamicConditions += `AND (c."settings"->>'disableCustomContributions')::boolean IS TRUE `;
     }
   }
 
   if (countryCodes) {
-    dynamicConditions += `AND "countryISO" IN (:countryCodes) `;
+    dynamicConditions += `AND (c."countryISO" IN (:countryCodes) OR parentCollective."countryISO" `;
```

Comment:
I don't get how this condition works (`OR parentCollective."countryISO"`? looks like we're missing something) plus it looks like there will be a syntax error for missing `)`.

It's super simple to add tests for this function in `test/server/lib/search.test.js`, let's make it right.

Reply:
Ops, I've mistakenly deleted part of the string. I've corrected it and added some tests as well. Thanks.

Reply:
Also I've created a new issue; https://github.com/opencollective/opencollective/issues/5530 to keep track of adding tests. I think we could add more tests around tags and type filters but we'll do it separately as this issue is only about country filter.